### PR TITLE
Homework #3 is done.

### PR DESCRIPTION
**Pull request by interist32**

### CODE REVIEW

File: hw-2/package.json [hunk 1]
```diff
@@ -9,17 +9,26 @@
     "test": "echo \"Error: no test specified\" && exit 1",
     "tsc": "tsc"
   },
+  "_moduleAliases": {
+    "@app-models": "build/models"
```
Comment (aledpardo) [inline hw-2/package.json]: Interesting! What's this for? Wasn't the change in `tsconfig.json` enough?

Comment (interist32) [reply hw-2/package.json]: Unfortunately it didn't work. 
Here's a separate PR where I enabled that feature: https://github.com/interist32/node.js-mentoring-program/pull/5

**Comment by aledpardo** [comment]
One more thing: consider to enable the `strictNullChecks` tsconfig option. It'll allow you to benefit in more depth of TS type checking capabilities.

File: hw-2/controllers/users.ts [hunk 2]
```diff
@@ -1,47 +1,45 @@
 import { Request, Response } from 'express';
 import { ValidatedRequest } from 'express-joi-validation';
-import UsersStorage from '../storage/users';
-import { UserRequestSchema, User } from '../models/user';
+import { UserRequestSchema, User } from '@app-models/user';
 
-const usersStorage = new UsersStorage();
+import UsersService from '../services/users';
+import UsersRepository from '../data-access/usersRepository';
```
[the remark this answers is not in the document]
Comment (interist32) [reply hw-2/controllers/users.ts]: Nice. It's my professional deformation. In Google Account we have different style guide, but thanks for reminding. I changed to users.service.ts like.

File: hw-2/data-access/usersRepository.ts [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+} from 'sequelize';
+import { User } from '@app-models/user';
+
+import uuid = require('uuid');
+
+// Hardcoded since my repo is private, and will be deleted after merge.
+const defaultConnectionString = 'postgres://iwsdzgkawdewtl:0652fa68f8c46ec0c712d0b253db84b871f1451a92698b7f20a7dee1f2a33691@ec2-54-92-174-171.compute-1.amazonaws.com:5432/d7c050r5hcsaas';
+const connectionString = process.env.CONNECTION_STRING || defaultConnectionString;
+
+export default class UsersRepository {
```
[the remark this answers is not in the document]
Comment (interist32) [reply hw-2/data-access/usersRepository.ts]: SGTM! Thanks!

File: hw-2/data-access/usersRepository.ts [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+      login: DataTypes.STRING,
+      password: DataTypes.STRING,
+      age: DataTypes.SMALLINT,
+      isDeleted: {
+        type: DataTypes.BOOLEAN,
+        defaultValue: false,
+      },
+    }, {
+      sequelize: this.sequelize,
+      tableName: 'users',
```
[the remark this answers is not in the document]
Comment (interist32) [reply hw-2/data-access/usersRepository.ts]: Fixed!

File: hw-2/data-access/usersRepository.ts [hunk 5]
```diff
@@ -0,0 +1,134 @@
+import {
+  Sequelize, DataTypes, Op, WhereAttributeHash,
+} from 'sequelize';
+import { User } from '@app-models/user';
+
+import uuid = require('uuid');
```
[the remark this answers is not in the document]
Comment (interist32) [reply hw-2/data-access/usersRepository.ts]: Done.

File: hw-2/controllers/users.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (aledpardo) [reply hw-2/controllers/users.ts]: Aha, so don't worry. Let's follow Google JS Style Guide then and keep it as is.

File: hw-2/controllers/users.ts [hunk 6]
```diff
@@ -1,47 +1,45 @@
...
+const usersService = new UsersService(new UsersRepository());
 
 export const getUsers = ({ query }: Request, res: Response): void => {
   const { login, limitQueryString } = query;
   const limit = +limitQueryString || undefined;
-  const users = usersStorage.getUsers(login, limit).map(prepareUser);
-  res.json(users);
+
+  usersService.getUsers(login, limit).then((users) => {
+    res.json(users.map((user) => user.toJSON()));
```
[the remark this answers is not in the document]
Comment (interist32) [reply hw-2/controllers/users.ts]: Thanks for explanation. Done.

File: hw-2/controllers/users.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (interist32) [reply hw-2/controllers/users.ts]: It's fine. I renamed it :)

**Pull request merged**